appinventor/appengine/src/com/google/appinventor/client/explorer/youngandroid/AssetList.java

```diff
@@ -95,15 +103,38 @@ public void onMouseMove(MouseMoveEvent event) {
         clientY = event.getClientY();
       }
     });
+    
+    
     assetList.addSelectionHandler(new SelectionHandler<TreeItem>() {
+      private int clickCount = 0;
       @Override
-      public void onSelection(SelectionEvent<TreeItem> event) {
-        TreeItem selected = event.getSelectedItem();
-        ProjectNode node = (ProjectNode) selected.getUserObject();
-        // The actual menu is determined by what is registered for the filenode
-        // type in CommandRegistry.java
-        ProjectNodeContextMenu.show(node, selected.getWidget(), clientX, clientY);
-      }});
+      public void onSelection(final SelectionEvent<TreeItem> event) {
+         clickCount++;
+         if (clickCount == 1) {
+            Timer timer = new Timer() {
+            @Override
+            public void run() {
+                if (clickCount == 1) {
+                   TreeItem selected = event.getSelectedItem();
+                   ProjectNode node = (ProjectNode) selected.getUserObject();
+                   // The actual menu is determined by what is registered for the filenode
+                   // type in CommandRegistry.java
+                   ProjectNodeContextMenu.show(node, selected.getWidget(), clientX, clientY);
+                }
+                if (clickCount == 2) {
```

Comment:
Turn this into an else-if clause since it is mutually exclusive with the previous if.